pkg/csiproxy/csi.go

```diff
@@ -125,13 +126,24 @@ func (p *Proxy) download() error {
 			return nil
 		},
 	}
+
+	// default to insecure
+	transport := http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
+	client.Transport = &transport
+
+	if !*p.cfg.Insecure && p.cfg.CertFilePath != "" {
+		transport = http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: false}}
+		client.Transport = &transport
+	}
+
 	resp, err := client.Get(fmt.Sprintf(p.cfg.URL, p.cfg.Version))
 	if err != nil {
 		return err
 	}
 
 	defer func(Body io.ReadCloser) {
 		_ = Body.Close()
+		transport.CloseIdleConnections()
```

Comment:
you should do a second defer up above as the client.Get could error and you might leak connections if you only have it in this defer